adobe/react-spectrum
# Update useLayoutEffect dependencies to ensure TabLine resizes

Adds two additional dependencies to the `useLayoutEffect` inside of `TabLine` that will ensure it's width is updated properly when the text inside the tab item changes.

Closes #2004 

## ✅ Pull Request Checklist:


## 📝 Test Instructions:
I attempted to write a new test to check that the inline style for `width` is updated correctly, however because the value is calculated using element offsets, those values always end up at `0` in a non-browser environment. So I added a new Storybook story instead to visualize the bugfix.

*Scenario 1*
- View new Storybook story for `Tabs -> Tab 1 controlled TabList item`
- Use the text field to adjust the first tab text
- Validate the `TabLine` grows/shrinks as expected

*Scenario 2*
- View new Storybook story for `Tabs -> Tab 1 controlled TabList item`
- Select Tab 2
- Use the text field to adjust the first tab text
- Validate the `TabLine` for Tab 2 moves horizontally to maintain alignment beneath Tab 2
